gcp-client/src/main/java/com/google/cloud/graphite/platforms/plugin/client/ContainerAnalysisClient.java

```diff
@@ -0,0 +1,189 @@
+
+  ContainerAnalysisClient(ContainerAnalysisWrapper containerAnalysis) {
+    this.containerAnalysis = containerAnalysis;
+  }
+
+  /**
+   * Wait until the vulnerability scan of the provided container image is complete and return the
+   * final status.
+   *
+   * @param projectId The ID of the project where the container image is hosted.
+   * @param resourceUrl The complete URL of the image in this form:
+   *     https://[HOST_NAME]/[PROJECT_ID]/[IMAGE_ID]@sha256:[HASH]
+   * @param timeoutMillis The number of milliseconds to wait for the scan before timing out.
+   * @return The final status of the scan, one of FINISHED_SUCCESS, FINISHED_FAILED, or
+   *     FINISHED_UNSUPPORTED.
+   * @throws ConditionTimeoutException If the the analysis was not in a finished state before the
+   *     timeout expired.
+   */
+  public String getVulnerabilityScanStatusSync(
+      final String projectId, final String resourceUrl, final long timeoutMillis)
+      throws ConditionTimeoutException {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(projectId));
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(resourceUrl));
+    Preconditions.checkArgument(timeoutMillis >= 0);
+    long start = System.currentTimeMillis();
+    Occurrence statusOccurence =
+        Awaitility.await()
+            .timeout(timeoutMillis, TimeUnit.MILLISECONDS)
+            .until(() -> getStatusOccurrence(projectId, resourceUrl), Optional::isPresent)
+            .get();
+    long timeLeft = timeoutMillis - (System.currentTimeMillis() - start);
+    return Awaitility.await()
+        .timeout(timeLeft, TimeUnit.MILLISECONDS)
+        .until(
+            () ->
+                getVulnerabilityScanStatus(projectId, nameFromSelfLink(statusOccurence.getName())),
+            Optional::isPresent)
+        .get();
+  }
+
+  /**
+   * Return a list of occurrences related to a vulnerability on the provided image.
+   *
+   * @param projectId The ID of the project where the container image is hosted.
+   * @param resourceUrl The complete URL of the image in this form:
+   *     https://[HOST_NAME]/[PROJECT_ID]/[IMAGE_ID]@sha256:[HASH]
+   * @return A list of {@link Occurrence}s of vulnerabilities on the image, sorted by name.
+   * @throws IOException An error occurred attempting to get the list of occurrences.
+   */
+  public ImmutableList<Occurrence> listVulnerabilityScanOccurrences(
+      final String projectId, final String resourceUrl) throws IOException {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(projectId));
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(resourceUrl));
+    return processResourceList(
+        containerAnalysis.listOccurrences(
+            projectId,
+            buildFilterString(
+                ImmutableMap.of("resourceUrl", resourceUrl, "kind", VULNERABILITY_KIND))),
+        Comparator.comparing(Occurrence::getName));
+  }
+
+  /**
+   * @param projectId The project ID where the container image is hosted.
```

Comment:
Missing method description. Also suggest making reference including a code example in the description demonstrating how this method would work with the CloudKMSClient.asymmetricSign() method.

Reply:
Done. Let me know what you think of the code sample. Doing this also helped me catch a weird case where the BinaryAuthorization API provides a method to generate an attestation payload but only through [gcloud](https://cloud.google.com/binary-authorization/docs/making-attestations#create_an_attestation_payload) or manually generating. It only requires the image's resource URL to generate so I added a method to do this in the BinaryAuthorizationClient as if the gcloud API call was exposed through the rest API, but let me know if you think there's a better place to put that.